$NetBSD: patch-CMakeLists.txt,v 1.5 2021/04/06 10:25:01 tnn Exp $

Do not make application bundle on Darwin.
Avoid
	CMake Error at CMakeLists.txt:3021 (install):
  	install FILES given directory
  	"${WRKDIR}/.buildlink/cmake-Modules" to install.

From 6ef6e8f607456e239e5f1a326803846930d34fa8 Mon Sep 17 00:00:00 2001
From: Gerald Combs <gerald@wireshark.org>
Date: Fri, 26 Mar 2021 12:52:01 -0700
Subject: [PATCH] CMake: Enable AUTO{MOC,UIC,RCC} earlier.

Enable CMAKE_AUTOMOC, CMAKE_AUTOUIC, and CMAKE_AUTORCC before searching
for Qt packages. This is apparently required for CMake 3.20.0 and later.
Fixes #17314.

(cherry picked from commit be2b0fc810fb2f2a80e471138dc29f0cbe87b209)

--- CMakeLists.txt.orig	2021-03-10 18:21:20.000000000 +0000
+++ CMakeLists.txt
@@ -1083,6 +1083,10 @@ ws_find_package(Systemd BUILD_sdjournal 
 
 # Build the Qt GUI?
 if(BUILD_wireshark)
+	set(CMAKE_AUTOMOC ON)
+	set(CMAKE_AUTOUIC ON)
+	set(CMAKE_AUTORCC ON)
+
 	# Untested, may not work if CMAKE_PREFIX_PATH gets overwritten
 	# somewhere. The if WIN32 in this place is annoying as well.
 	if(WIN32)
@@ -1388,7 +1392,7 @@ else ()
 	set(EXTCAP_INSTALL_LIBDIR "${CMAKE_INSTALL_LIBDIR}/${PROJECT_NAME}/extcap" CACHE INTERNAL "The extcap dir")
 endif()
 
-if(APPLE)
+if(APPLE AND ENABLE_APPLICATION_BUNDLE)
 	#
 	# As https://developer.apple.com/library/archive/technotes/tn2206/_index.html
 	# says,
@@ -2345,7 +2349,11 @@ if(BUILD_wireshark AND QT_FOUND)
 		${MINIZIP_LIBRARIES}
 	)
 
-	add_executable(wireshark WIN32 MACOSX_BUNDLE ${wireshark_FILES} ${EXTRA_BUNDLE_FILES})
+	if(ENABLE_APPLICATION_BUNDLE)
+		add_executable(wireshark WIN32 MACOSX_BUNDLE ${wireshark_FILES} ${EXTRA_BUNDLE_FILES})
+	else()
+		add_executable(wireshark WIN32 ${wireshark_FILES} ${EXTRA_BUNDLE_FILES})
+	endif()
 	if(WIN32)
 		set_property(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} PROPERTY VS_STARTUP_PROJECT wireshark)
 	endif()
@@ -3284,12 +3292,12 @@ endif()
 
 install(
 	FILES
-		${CMAKE_MODULE_PATH}/FindGLIB2.cmake
-		${CMAKE_MODULE_PATH}/FindWSLibrary.cmake
-		${CMAKE_MODULE_PATH}/FindWSWinLibs.cmake
-		${CMAKE_MODULE_PATH}/UseAsn2Wrs.cmake
-		${CMAKE_MODULE_PATH}/LocatePythonModule.cmake
-		${CMAKE_MODULE_PATH}/UseMakePluginReg.cmake
+		cmake/modules/FindGLIB2.cmake
+		cmake/modules/FindWSLibrary.cmake
+		cmake/modules/FindWSWinLibs.cmake
+		cmake/modules/UseAsn2Wrs.cmake
+		cmake/modules/LocatePythonModule.cmake
+		cmake/modules/UseMakePluginReg.cmake
 	DESTINATION
 		${WIRESHARK_INSTALL_CMAKEDIR}
 )
